types/device.go

```diff
@@ -27,6 +27,7 @@ type DeviceRequest struct {
 	Driver       string      `yaml:"driver,omitempty" json:"driver,omitempty"`
 	Count        DeviceCount `yaml:"count,omitempty" json:"count,omitempty"`
 	IDs          []string    `yaml:"device_ids,omitempty" json:"device_ids,omitempty"`
+	Options      Mapping     `yaml:"options,omitempty" json:"options,omitempty"`
```

Comment:
We are not doing anything with this yet. Should we wait for the upcoming work to add this?

Reply:
we are not because this was missing in compose-go, but [moby API](https://docs.docker.com/reference/api/engine/version/v1.45/#tag/Container/operation/ContainerCreate) has support for options (sorry, can't post a direct link to DeviceRequest nested struct)
> Driver-specific options, specified as a key/value pairs. These options are passed directly to the driver.

Reply:
Apologies, re-reading my comment I realise I did not express myself correctly. I meant, in compose-go you defined the `Options` but not doing any actual implementation with it. All I am saying is should we save this change for later for when we augment the options section or should we already implement the moby api call in this PR?

Reply:
compose-go never includes implementation; this is docker/compose role to do so

Reply:
indeed, thanks

Reply:
fixed, +added a test case to cover loading DeviceRequest

Reply:
@ndeloof @glours thanks!